tests/testthat/test-pre_process.R

```diff
@@ -0,0 +1,96 @@
+
+testthat::test_that("get_month returns correct month",{
+  expect_match(class(get_month(1)), "character")
+
+  expect_match(get_month(1), "Jan")
+  expect_match(get_month(1), "Jan")
+  expect_match(get_month(1.5), "Jan")
+  expect_equal(get_month(13), "NA")
+  expect_equal(get_month("Jan"), "NA")
+  expect_equal(get_month(NA), "NA")
+})
+
+testthat::test_that("format_flowering_months returns a single 12 character string",{
+  expect_match(class(format_flowering_months(1, 1)), "character")
+  expect_equal(stringr::str_length(format_flowering_months(1, 1)), 12)
+  
+  expect_match(format_flowering_months(1, 6), "yyyyyynnnnnn")
+  expect_match(format_flowering_months(1, 12), "yyyyyyyyyyyy")
+  expect_match(format_flowering_months(12, 12), "nnnnnnnnnnny")
+  expect_equal(format_flowering_months(12, 6), "yyyyyynnnnny")
+  expect_equal(format_flowering_months(0, 6), "NA")
+  expect_equal(format_flowering_months(1, 13), "NA")
+  expect_equal(format_flowering_months(12, 13), "NA")
+  expect_equal(format_flowering_months(1, NA), "NA")
+})
+
+testthat::test_that("convert_month_range_vec_to_binary returns a single 12 character string",{
+  expect_match(class(convert_month_range_vec_to_binary("Mar")), "character")
+  expect_equal(stringr::str_length(convert_month_range_vec_to_binary("Mar")), 12)
+  
+  expect_match(convert_month_range_vec_to_binary("Mar"), "nnynnnnnnnnn")
+  expect_match(convert_month_range_vec_to_binary("Mar-Apr"), "nnyynnnnnnnn")
+  expect_match(convert_month_range_vec_to_binary("Nov-Apr"), "yyyynnnnnnyy")
+  expect_match(convert_month_range_vec_to_binary("nov-apr"), "yyyynnnnnnyy")
+  expect_match(convert_month_range_vec_to_binary("Feb-Mar;Oct-Nov"), "nyynnnnnnyyn")
+  expect_equal(convert_month_range_vec_to_binary(c("Mar", "Apr")), c("nnynnnnnnnnn","nnnynnnnnnnn"))
+  expect_equal(convert_month_range_vec_to_binary("March"), "NA")
+  expect_equal(convert_month_range_vec_to_binary(3), "NA")
+  expect_equal(convert_month_range_vec_to_binary(NA), "NA")
+})
+
+testthat::test_that("convert_01_ny returns correct character",{
+  expect_match(class(convert_01_ny(1)), "character")
+  
+  expect_match(convert_01_ny(0), "n")
```

Comment:
Great! Should we also test for sequences of 01?

Also, I think inputs will be a character, so perhaps test some character inputs too?

Reply:
Good idea, just tried testing 00 and 01. 00 and 01 return "n" and "y" rather then "nn" and "ny", it seems to ignore the leading 0. When "00" and "01" are used are used as the input it returns "nn" and "ny" correctly. The same behaviour occurs if you have a bunch of leading 0s followed by a one. All of the 0s are ignored.

Reply:
Yes, really only makes sense to pass values as characters. So I suggest changing tests to use character strings rather than numeric